server/channels/store/sqlstore/retention_policy_store.go

```diff
@@ -1152,9 +1152,16 @@ func getDeleteQueriesForMySQL(r RetentionPolicyBatchDeletionInfo, query string)
 	return fmt.Sprintf("DELETE %s FROM %s INNER JOIN (%s) AS A ON %s", r.Table, r.Table, query, joinClause)
 }
 
-func deleteFromRetentionIdsTx(txn *sqlxTxWrapper, id string) (err error) {
-	if _, err := txn.Exec("DELETE FROM RetentionIdsForDeletion WHERE Id = ?", id); err != nil {
-		return errors.Wrap(err, "Failed to delete from RetentionIdsForDeletion")
+func deleteFromRetentionIdsTx(txn *sqlxTxWrapper, id string) (int64, error) {
+	sqlResult, err := txn.Exec("DELETE FROM RetentionIdsForDeletion WHERE Id = ?", id)
+	if err != nil {
+		return 0, errors.Wrap(err, "Failed to delete from RetentionIdsForDeletion")
 	}
-	return nil
+
+	rowsAffected, err := sqlResult.RowsAffected()
+	if err != nil {
+		return 0, errors.Wrapf(err, "unable to retrieve rows affected")
+	}
```

Comment:
Just one question: Is it possible to fail here and return an error when the deletion succeeded?

Reply:
Maybe another: in Calls we had problems with `RowsAffected` having implementation-specific behaviour, is there a danger of that here?

Reply:
Good questions @cpoile. I'm not sure if I have the answers to these.


> Maybe another: in Calls we had problems with `RowsAffected` having implementation-specific behaviour, is there a danger of that here?

`RowsAffected` is used in the code base without differentiating between different driver implementations. Would you happen to have a handy link to the conversations about this in Calls so I can take a deeper look?



> Just one question: Is it possible to fail here and return an error when the deletion succeeded?

In all other places of the codebase where `RowsAffected` is used, the method returns with an error if RowsAffected` errors. I have never seen `RowsAffected` fail. So, I assume we are as safe here as we are with other parts of the code base.

Reply:
Thanks for sharing that additional piece of context @cpoile. It seems like we are good here, right?